utilities/write_batch_with_index/write_batch_with_index.cc

```diff
@@ -216,9 +218,19 @@ class BaseDeltaIterator : public Iterator {
     }
     // equal_keys_ <=> compare == 0
     assert((equal_keys_ || compare != 0) && (!equal_keys_ || compare == 0));
+
 #endif
   }
 
+  void IsOverUpperBound() {
+    if ((read_options_.iterate_upper_bound != nullptr &&
+         comparator_->Compare(key(),
+                              *read_options_.iterate_upper_bound) >= 0)) {
+      current_over_upper_bound_ = true;
```

Comment:
I don't see this is reset to false when people calling Seek() or similar call.

Reply:
ah! I should make `IsOverUpperBound()` return a boolean and update `current_over_upper_bound_` at the end of `UpdateCurrent()`